### PR TITLE
LPS-171574 | Clarify oneToMany fields information

### PR DESCRIPTION
**Pull request by Hijky**
**Background**
Add a testcase to clarify one to many fields information

**Test Plan**
**Background Steps:**
Given object definitions Student, Subject, University with text field name created
And Given universityStudents oneToMany relationship is created
And Given subjectStudents oneToMany relationship is created
And Given a Student entry, Subject entry, University entry created
And Given University entry related to the Student entry
And Given Student entry related to the Subject entry

**CanGetMultipleNestedFieldsInOneToManyRelationshipFromChild** 
When requesting GET /o/c/students/{studentId}?nestedFields=universityStudents,subjectStudents"
Then nodes subjectStudents,universityStudents contain related student entry data

**CanGetNestedFieldsNodeInOneToManyRelationshipFromParent**
When requesting GET /o/c/subjects/{subjectId}?nestedFields=subjectStudents
Then subjectStudents node contains related student entry data

**CanGetRelatedObjectEntryIdInOneToManyRelationshipFromChild**
When requesting GET /o/c/students/{studentId}
Then the response contains r_universityStudents_c_universityId,r_universityStudents_c_universityERC,universityStudentsERC values

**JIRA Link:**
https://issues.liferay.com/browse/LPS-171574

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:relevant

**Comment by Hijky** [comment]
ci:test:headless-functional

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Review by yanan — APPROVED** [review]

File: modules/apps/object/object-rest-test/src/testFunctional/tests/ClarifyOneToManyFieldsInformation.testcase [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		task ("When requesting GET /o/c/students/{studentId}?nestedFields=universityStudents,subjectStudents") {
+			var response = ObjectDefinitionAPI.getObjectEntryById(
+				en_US_plural_label = "students",
+				objectEntryId = "${staticStudentId}",
+				parameter = "nestedFields",
+				parameterValue = "universityStudents,subjectStudents");
+		}
+
+		task ("Then in r_universityStudents_c_university node the value r_universityStudents_c_university.id and r_universityStudents_c_universityId have the value of universityId") {
+			var actualId = JSONUtil.getWithJSONPath("${response}", "$..r_universityStudents_c_university[?(@.id==${staticUniversityId})].id");
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/tests/ClarifyOneToManyFieldsInformation.testcase:91]: Nice! :+1:

File: modules/apps/object/object-rest-test/src/testFunctional/tests/ClarifyOneToManyFieldsInformation.testcase [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		for (var objectName : list "University,Student,Subject") {
+			ObjectAdmin.deleteObjectViaAPI(objectName = "${objectName}");
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = "3"
+	test CanGetMultipleNestedFieldsInOneToManyRelationship {
+		property portal.acceptance = "true";
+
```
Comment (magjed4289) [inline modules/apps/object/object-rest-test/src/testFunctional/tests/ClarifyOneToManyFieldsInformation.testcase:81]: property portal.acceptance = "true"; not needed in FP3 tasks :)

**Review by magjed4289 — CHANGES_REQUESTED** [review]

**Comment by magjed4289** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:relevant

**Comment by Hijky** [comment]
ci:test:headless-functional

**Comment by Hijky** [comment]
@magjed4289 @yanan The failures are not related to my PR, and if the [ticket](https://issues.liferay.com/browse/LPS-172697) has been solved I will update the PR.

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Review by magjed4289 — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Review by Hijky — COMMENTED** [review]
Looks good! A perfect macro objectUntils for us.

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:relevant

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by magjed4289** [comment]
ci:forward

**Pull request closed without merging**